service_configuration_lib/spark_config.py

```diff
@@ -31,10 +32,20 @@
 GPUS_HARD_LIMIT = 15
 CLUSTERMAN_METRICS_YAML_FILE_PATH = '/nail/srv/configs/clusterman_metrics.yaml'
 CLUSTERMAN_YAML_FILE_PATH = '/nail/srv/configs/clusterman.yaml'
+
+MEDIUM_EXECUTOR_CORES = 8
```

Comment:
All these global variables look too crowded now.
Maybe we can store them in dictionary for easy modification and code searchability easily?
Something like:
# Probably some better variable name
RECOMMENDED_SPARK_RESOURCE_CONFIGS = {
  'executor_cores': {
     'medium': 8
     'max': 12
     'recommended': 4
 },
 'executor_memory': {
   'recommended': 28
   'medium': 56
   'max': 112
},
'target_mem_core_ratio': 7
}.
Or even open to some better data-structure.
What do you think?

Reply:
How about adding a dict with namedtuple `ExecutorResourceConfig` to store the values?